src/extension/providers/providerBase.ts

```diff
@@ -8,11 +8,20 @@ export default class ProviderBase {
   private _scope: string;
 
   constructor(scope: string) {
-    this.enabled = false;
+    this.enabled = this.loadEnabledState();
     this._eventEmitter = new EventEmitter();
     this._scope = scope;
   }
 
+  private loadEnabledState(): boolean {
```

Comment:
@pavanjoshi914 we were planning to add an `isEnabled()` function right? can this be added to the same PR?